### PR TITLE
Add initial support for P2WSH inputs - features/transactions

### PR DESCRIPTION
**Pull request by jakinyele**

### CODE REVIEW

File: bitcoin/src/address.rs [hunk 1]
```diff
@@ -70,6 +73,29 @@ impl<N: BitcoinNetwork> BitcoinAddress<N> {
         })
     }
 
+    // Computes hash and returns a Bech32 segwit address from a script
```
Comment (howardwu) [inline bitcoin/src/address.rs]:
```suggestion
    // Returns a P2WSH address in Bech32 format from a given Bitcoin script.
```

File: bitcoin/src/address.rs [hunk 2]
```diff
@@ -70,6 +73,29 @@ impl<N: BitcoinNetwork> BitcoinAddress<N> {
...
+    pub fn p2wsh(original_script: &Vec<u8>) -> Result<Self, AddressError> {
+        let script2 = Sha256::digest(&original_script).to_vec();
+
+        // organize as a hash
```
Comment (howardwu) [inline bitcoin/src/address.rs]:
```suggestion
        // Organize as a hash
```

File: bitcoin/src/address.rs [hunk 3]
```diff
@@ -70,6 +73,29 @@ impl<N: BitcoinNetwork> BitcoinAddress<N> {
...
+        let mut redeem_script = vec![0x00, 0x20];
+        redeem_script.extend(script2.iter());
+
+        let version = u5::try_from_u8(redeem_script[0])?;
+
+        let mut data = vec![version];
+        // get the SHA256 hash of the script
```
Comment (howardwu) [inline bitcoin/src/address.rs]:
```suggestion
        // Get the SHA256 hash of the script
```

File: bitcoin/src/format.rs [hunk 4]
```diff
@@ -10,6 +10,8 @@ use std::fmt;
 pub enum BitcoinFormat {
     /// Pay-to-Pubkey Hash, e.g. 1NoZQSmjYHUZMbqLerwmT4xfe8A6mAo8TT
     P2PKH,
+    /// Pay-to-Script Hash, e.g. 347N1Thc213QqfYCz3PZkjoJpNv5b14kBd
```
Comment (howardwu) [inline bitcoin/src/format.rs]:
```suggestion
    /// Pay-to-Witness-Script Hash, e.g. 347N1Thc213QqfYCz3PZkjoJpNv5b14kBd
```

File: bitcoin/src/transaction.rs [hunk 5]
```diff
@@ -245,6 +253,15 @@ impl<N: BitcoinNetwork> Outpoint<N> {
                             false => None,
                         },
                     },
+                    BitcoinFormat::NATIVE_P2WSH => match redeem_script {
+                        Some(redeem_script) => match script_pub_key[0] != 0x00 as u8
+                            && script_pub_key[1] != 0x20 as u8 && script_pub_key.len() != 34 // zero [32-byte sha256(witness script)]
+                        {
+                            true => return Err(TransactionError::InvalidScriptPubKey("NATIVE_P2WSH".into())),
```
Comment (howardwu) [inline bitcoin/src/transaction.rs]: Nit: Friendly reminder to change instances of text containing `"NATIVE_P2WSH"` to `"P2WSH"` as well when making the variable name change.

File: bitcoin/src/address.rs [hunk 6]
```diff
@@ -11,6 +11,7 @@ use wagyu_model::{
 use base58::{FromBase58, ToBase58};
 use bech32::{u5, Bech32, FromBase32, ToBase32};
 use std::{convert::TryFrom, fmt, marker::PhantomData, str::FromStr};
+use sha2::{Digest, Sha256};
```
[the remark this answers is not in the document]
Comment (jakinyele) [reply bitcoin/src/address.rs]: Thanks for the feedback. Fixing!

**Pull request merged**